gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -263,6 +266,24 @@ private Instant getCutoffTimeForRef(
         ? gcParams.getDefaultCutOffTimestamp()
         : gcParams
             .getCutOffTimestampPerRef()
-            .getOrDefault(reference.getName(), gcParams.getDefaultCutOffTimestamp());
+            .getOrDefault(reference, gcParams.getDefaultCutOffTimestamp());
+  }
+
+  private static Reference deSerializeReference(String reference) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      return objectMapper.readValue(reference, Reference.class);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static String serializeContent(Content content) {
+    ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
`objectMapper` should rather be kept in a static final. Also I think it would make sense to have this functionality in a utility class or something so that I can be reused (rather than copied)

Reply:
Moved to `GCUtil`